server/graphql_schemas/event/schema.py

```diff
@@ -1,65 +1,107 @@
+        title = graphene.String(required=True)
+        description = graphene.String(required=True)
+        venue = graphene.String(required=True)
+        date = graphene.String(required=True)
+        time = graphene.String(required=False)
+        featured_image = graphene.String(required=False)
+        social_event_id = graphene.String(required=False)
 
     new_event = graphene.Field(EventNode)
 
     @classmethod
     def mutate_and_get_payload(cls, root, info, **input):
         social_event_id = input.get('social_event_id')
-        social_event = Category.objects.get(id=int(social_event_id))
+        social_event = Category.objects.get(
+            id=from_global_id(social_event_id)[1])
         user = info.context.user
-        googleUser = GoogleUser.objects.get(app_user_id=user.id)
+        googleUser = GoogleUser.objects.first()
         new_event = Event(
-          title=input.get('title'),
-          description=input.get('description'),
-          venue=input.get('venue'),
-          date=input.get('date'),
-          time=input.get('time'),
-          featured_image=input.get('featured_image'),
-          creator=googleUser,
-          social_event=social_event
+            title=input.get('title'),
+            description=input.get('description'),
+            venue=input.get('venue'),
+            date=input.get('date'),
+            time=input.get('time'),
+            featured_image=input.get('featured_image'),
+            creator=googleUser,
+            social_event=social_event
         )
         new_event.save()
 
         return cls(new_event=new_event)
 
 
-class EventQuery(object):
-  event = graphene.Field(EventNode,
-                         id=graphene.Int(),
-                         title=graphene.String())
-  events_list = DjangoFilterConnectionField(EventNode)
+class UpdateEvent(relay.ClientIDMutation):
+
+    class Input:
+        title = graphene.String()
+        description = graphene.String()
+        venue = graphene.String()
+        date = graphene.String()
+        time = graphene.String()
+        featured_image = graphene.String()
+        social_event_id = graphene.Int()
+        event_id = graphene.String(required=True)
+
+    updated_event = graphene.Field(EventNode)
+    action_message = graphene.String()
+
+    def update_instance(instance, args, exceptions=['event_id']):
+        if instance:
+            [setattr(instance, key, value)
+             for key, value in args.items() if key not in exceptions]
+            instance.save()
+        return instance
+
+    @classmethod
+    def mutate_and_get_payload(cls, root, info, **input):
+
+        try:
+            user = GoogleUser.objects.get(app_user=info.context.user)
+            event_instance = Event.objects.get(
+                id=from_global_id(input.get('event_id'))[1]
+            )
+            if event_instance.creator != user \
+                    and not info.context.user.is_superuser:
```

Comment:
Well done @parseendavid But please why are we checking if the user is not superuser on line 84?

Reply:
@tosmak16 the intepretation of the line of code is that if a user is not the creator of the event and is not an admin/superuser then the user is not allowed to edit an event.

Reply:
Okay...Thanks.